endaq/batch/core.py

```diff
@@ -505,27 +505,45 @@ def _get_data(self, filename):
 
         return data
 
-    def aggregate_data(self, filenames) -> OutputStruct:
+    def aggregate_data(self, filenames) -> Optional[OutputStruct]:
         """
         Compile configured data from the given files into a dataframe.
 
         :param filenames: a sequence of paths of recording files to process
         """
-        filenames = [os.path.abspath(name) for name in filenames]
-        root_path = os.path.commonpath(filenames)
-        file_display_names = [
-            os.path.relpath(name, start=root_path) for name in filenames
+        if len(filenames) == 0:
+            return None
+
+        http_files, local_files = [], []
+        for file in filenames:
+            (
+                http_files
+                if file.startswith("http://") or file.startswith("https://")
```

Comment:
@StokesMIDE is this an acceptable check for whether a file is a network or local file?

Reply:
It's probably fine, although the filename *could* be a `file:` URL, which would need parsing... you might want to consider something like what I did in `get_doc()`: https://github.com/MideTechnology/endaq-python/blob/d2196257c96ce6e0297d01de33c65dbd1c8340ea/endaq/ide/files.py#L157

Reply:
File URLs can be generated if a filename is copied to/from an email or something, so it isn't entirely unlikely.

Reply:
yeah I considered it, but I wasn't sure how deep I wanted to get into the issue. how about a7c3b456aaf8ee8a21ea4be84a8b7c84b060264e?

Reply:
@StokesMIDE I got an approval, but is there anything you wanted to discuss here?